src/main/java/jenkins/plugins/openstack/compute/slaveopts/BootSource.java

```diff
@@ -354,6 +354,21 @@ public String toString() {
                 return "Volume from Image";
             }
         }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o)
+                return true;
+            if (o == null || getClass() != o.getClass())
+                return false;
+            final VolumeFromImage that = (VolumeFromImage) o;
+            return name.equals(that.name);
```

Comment:
This doesn't check all fields.
Use the super.equals() to check the parent class's fields and, if that returns true, then check this class's additions ((volumeSize).